core/src/repair/serve_repair.rs

```diff
@@ -526,23 +526,26 @@ impl ServeRepair {
         }
         Self::verify_signed_packet(my_id, &remote_request.bytes, &request)?;
         if let Some(remote_pubkey) = remote_request.remote_pubkey {
-            if &remote_pubkey != request.sender() {
+            if Some(&remote_pubkey) != request.sender() {
```

Comment:
nit: Certainly don't need to address it as part of this PR, but wondering if we should just call `request.send()` once given that it performs an underlying match.

Reply:
+1

Reply:
This is only a sanity check on the QUIC path of repair.
We shouldn't hit the `error!` line anyways.

Reply:
I see 5 including the error line, so 4 excluding it:
https://github.com/anza-xyz/agave/blob/680946c77a4f3b24bcd4f006cc3020d4c933827e/core/src/repair/serve_repair.rs#L529
https://github.com/anza-xyz/agave/blob/680946c77a4f3b24bcd4f006cc3020d4c933827e/core/src/repair/serve_repair.rs#L532
https://github.com/anza-xyz/agave/blob/680946c77a4f3b24bcd4f006cc3020d4c933827e/core/src/repair/serve_repair.rs#L536
https://github.com/anza-xyz/agave/blob/680946c77a4f3b24bcd4f006cc3020d4c933827e/core/src/repair/serve_repair.rs#L542
https://github.com/anza-xyz/agave/blob/680946c77a4f3b24bcd4f006cc3020d4c933827e/core/src/repair/serve_repair.rs#L545-L546

Completely unrelated, but I was not aware you could use `?` on an `Option` as line 542 does ... today I learned

Reply:
> I see 5 including the error line, so 4 excluding it:

Oh, I thought you just mean it in that `if` branch.